agent/agent_test.go

```diff
@@ -19,6 +22,54 @@ func TestAgent_OmitHostname(t *testing.T) {
 	assert.NotContains(t, c.Tags, "host")
 }
 
+type testIgnoreErrorInput struct {
+}
+
+func (i *testIgnoreErrorInput) Init() error {
+	return fmt.Errorf("could not initialize input: test error")
+}
+
+func (i *testIgnoreErrorInput) Gather(telegraf.Accumulator) error {
+	return nil
+}
+func (i *testIgnoreErrorInput) SampleConfig() string {
+	return ""
+}
+
+func TestAgent_IgnoreErrorInputs(t *testing.T) {
+	c := config.NewConfig()
+	assert.False(t, c.Agent.IgnoreErrorInputs)
+	inputs := make([]*models.RunningInput, 0)
+	inputs = append(inputs, &models.RunningInput{})
+	c.Inputs = inputs
+	a, err := NewAgent(c)
+	assert.NoError(t, err)
+	a.initPlugins()
+	assert.Equal(t, 1, len(c.Inputs))
+
+	errorInputs := make([]*models.RunningInput, 0)
+	errorInput := &models.RunningInput{
+		Config: &models.InputConfig{
+			Name:     "test error input",
+			Alias:    "test alias",
+			Interval: 10 * time.Second,
+		},
+		Input: &testIgnoreErrorInput{},
+	}
+	errorInputs = append(errorInputs, errorInput)
+	c.Inputs = errorInputs
```

Comment:
the following code suggestion just gives the idea, it might not can be compiled !
```suggestion
	c.Inputs = []*models.RunningInput {
		Config: &models.InputConfig{
			Name:     "test error input",
			Alias:    "test alias",
			Interval: 10 * time.Second,
		},
		Input: &testIgnoreErrorInput{},
	}
```